### DISCUSSION TITLE
v1.0.0-preview.1

### DISCUSSION THREAD

**Discussion by ryanbliss** [microsoft/live-share-sdk#112] [Announcements]
This release brings major breaking changes as we prepare to bring Live Share to v1.0.0. Please carefully review before updating your app.

### Changes to @microsoft/live-share

**Major changes**
- Renamed `TeamsFluidClient` to `LiveShareClient`
- Added `LiveShareHost` interface as parameter when initializing`LiveShareClient` to override default behavior in Microsoft Teams container management, shared clock, etc.
- Renamed `EphemeralPresence` to `LivePresence`
- Renamed `EphemeralEvent` to `LiveEvent`
- Renamed `EphemeralState` to `LiveState`
- Renamed `EphemeralTimer` to `LiveTimer`
- Added public `testLiveShare` namespace for joining a container locally in a browser

### Changes to @microsoft/live-share-media

**Major changes**
- Renamed `EphemeralMediaSession` to `LiveMediaSession`
- Renamed `EphemeralMediaSessionCoordinator` to `LiveMediaSessionCoordinator`
- Renamed `VolumeLimiter` to `VolumeManager`
- `VolumeManager` now has `startLimiting` and `stopLimiting` to more closely align with modeling of teams-js API for `registerSpeakingStateChangeHandler`
- `VolumeManager` renamed `level` to `limitLevel`
- `VolumeManager` renamed `limitType` to `limitLevelType`
- `VolumeManager` now supports setting the user selected volume with a `volume` property for compatibility with volume sliders. When limiting ends, the user is returned back to the set `volume` level.

**Minor changes**
- Play/pause at zero seconds when in view only mode now properly blocks the action

### New package: @microsoft/live-share-canvas

This was first released in v0.4.1, which we never prepared release notes for. This package allows applications to add turn-key inking and cursors in their UI. For more information, visit our concept docs at https://aka.ms/livesharecanvas, or try out any of the following samples:
- 03.live-canvas-demo
- 21.react-meda-template
- 23.react-live-canvas

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/live-share-sdk/releases/tag/v1.0.0-preview.1'>v1.0.0-preview.1</a>.</em>

**Comment by CaptainTammy** [comment]
@ryanbliss 
Is there any plan or timeline yet when this will be out of preview and production released?

**Reply by ryanbliss** [reply]
Hi @CaptainTammy good question. Live Share is available in our public "rings" in Teams, and internally at Microsoft is viewed as production ready. The YouTube application in the Teams Store (developed by Microsoft) uses Live Share v1.0.0-preview.4. 

The only thing holding us back at this point is that we are still weighing a few minor breaking changes to our interfaces before fully shipping to v1.0.0, after which we can't make any more breaking changes until v2. At the latest, we will publish to v1 to npm by Build 2023, which is in May. We are hoping to get it out before then. 

We'll publish an update soon. Hopefully this helps!
